Sources/xcodeproj/Utils/ReferenceGenerator.swift

```diff
@@ -25,6 +27,16 @@ final class ReferenceGenerator: ReferenceGenerating {
         }
 
         self.proj = proj
+
+        // cache current reference values
+        var references: Set<String> = []
```

Comment:
What about creating an extension for collections whose type is `PBXObject`? We could extract this logic there and simplify this code by doing:

```swift
self.references = proj.objects.fixedReferences
```


Moreover, what about renaming `references` to `cachedReferences`?

Reply:
> Objects isn’t a collection. Then we’d need a getObjects() function. Could add it, just trying to cut down on those huge methods that reference every type.

Fair enough :+1:

> Sure it could be renamed. This is all internal stuff so can be refactored. I just thought it’s more than a cache and is actually used in the logic

Let's keep it as `references`.